Sources/SymbolKit/SymbolGraph/SemanticVersion/SemanticVersion.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 - 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension SymbolGraph {
+    /// A [semantic version](https://semver.org).
+    public struct SemanticVersion {
+        /// The major version.
+        public let major: UInt
+        /// The minor version.
+        public let minor: UInt
+        /// The patch version.
+        public let patch: UInt
+        /// Dot-separated pre-release identifiers.
+        public var prereleaseIdentifiers: [String] { prerelease.identifiers.map(\.description) }
```

Comment:
Could you elaborate a bit on the reasoning for preferring optionals here?

The reason I have chosen arrays is because by my understanding the semantic version string encodes arrays of identifiers at the pre-release and build metadata positions.

Reply:
To clarify: I meant that I feel that it would be more correct to model prerelease information and build metadata as optional array of strings (`[String]?`). That way, if the version didn't have any prerelease information it would be `nil` but if it had more than one piece of prerelease information than those would be values in the array.

That change would mean that `Prerelease` doesn't need to support initializing with a nil value and places that currently do `isEmpty` checks get replaced with `nil` checks.

Reply:
Thanks for the clarification.

> That change would mean that `Prerelease` doesn't need to support initializing with a nil value

This combined with the other comment, do you mean to avoid taking an optional argument by initializing like this:

```swift
prereleaseStorage = prereleaseSubstring.map { Prerelease(String($0)) }
```

If this is the case, then I think it makes more sense to me now. Is there any advantage for doing it this way (other than for stylistic preferences)? So far it feels to me like an additional layer of expressing emptiness that an empty array is already capable of.

Reply:
It is essentially a stylistic preference. 

The only difference would be at the call site where an optional value needs to be handled explicitly.